internal/vault/client.go

```diff
@@ -163,6 +169,50 @@ type defaultClient struct {
 	mu                 sync.RWMutex
 }
 
+func (c *defaultClient) IsClone() bool {
+	return c.isClone
+}
+
+func (c *defaultClient) Namespace() string {
+	return c.client.Namespace()
+}
+
+func (c *defaultClient) SetNamespace(s string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.client.SetNamespace(s)
+}
+
+func (c *defaultClient) Clone(namespace string) (Client, error) {
+	if namespace == "" {
+		return nil, errors.New("namespace cannot be empty")
+	}
+
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	clone, err := c.client.Clone()
+	if err != nil {
+		return nil, err
+	}
+	clone.SetToken(c.client.Token())
```

Comment:
Does setting `config.CloneToken = true` in MakeVaultClient() not actually clone the token?

Reply:
Hmm, I don't see 1a28323f25fc18e3791844589c865f76376b9dde here yet.

Reply:
Oops, should be there now. Forgot to push it :(